src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -303,6 +398,18 @@ class WordHighlighter {
 	}
 
 	private _removeDecorations(): void {
+		const currentEditors = this.codeEditorService.listCodeEditors();
+		// iterate over editors and store models in currentModels
+		for (const editor of currentEditors) {
+			const model = editor.getModel();
+			if (model) {
+				model.changeDecorations((changeAccessor) => {
+					changeAccessor.deltaDecorations(this.storedDecorations.get(model.uri) ?? [], []);
```

Comment:
Important: You cannot use `model.uri` with just `Map` because the map works with instance identify. It's likely the instance you want but there is no gurantee. Either use `...uri.toString()` or (better) use `ResourceMap` which does this for you

Reply:
nit - Also, consider to check `this.storedDecorations.get(model.uri)` before entering `changeDecorations`